application/classes/Ushahidi/Repository/Form/Contact.php

```diff
@@ -260,6 +257,19 @@ public function existsInActiveTargetedSurvey($contact)
 		}
 	}
 
+
+	/**
+	 * @param int $contact_id
+	 * @param int $form_id
+	 * @return bool
+	 */
+	public function setInactiveTargetedSurvey($tss_id, $form_id)
+	{
+		$repo = $this->targeted_survey_state_repo->get($tss_id);
+		$entity = $repo->setState(array('survey_status' => str_replace('###', $form_id,Entity\TargetedSurveyState::INVALID_CONTACT_MOVED)));
```

Comment:
Nice work with adding to the constant; I didn't realize how complicated that would be when I suggested it.